common/src/main/scala/explore/common/SSOClient.scala

```diff
@@ -75,11 +98,45 @@ object SSOClient {
               .getOrElse(throw new RuntimeException("Error decoding the token"))
           }
         case Response(Left(e), _, _, _, _, _)     =>
-          Sync[F].raiseError(new RuntimeException(e))
+          throw new RuntimeException(e)
+      }
+  }
+
+  private def whoami[F[_]: Sync](
+    fromFuture: FromFuture[F, Response[Either[String, String]]]
+  ): F[Option[UserVault]] = {
+    val backend  = FetchBackend(FetchOptions(RequestCredentials.include.some, none))
+    val httpCall =
+      Sync[F].delay(
+        basicRequest
+          .post(
+            uri"$baseSSO/api/v1/refresh-token"
```

Comment:
The `baseSSO` can go into `AppConfig` now.

Reply:
Yes, I have that ready to go in the next PR